### PR TITLE
feat: Add events page

### LINKED ISSUE #20: Events Page
Create sections required for the events page.

### Tasks:

**Sections**

- [ ] Friday Night Games
- [ ] Duck Lounge
- [ ] Image carousel of images from different events
- [ ] Upcoming Events

**Components**

- [ ] Events list (method to be determined, either use an API to get events from our Facebook page or manually add to the database)

### PR DESCRIPTION
**Pull request by jsun969**
### Description
Add events page

### Changes Made
- Events page
- Easter egg
- Fix prettier config
- Minor fix

### Related Issues
Resolves #20

### Additional Notes
- Carousel images and events data need to be updated in `data/events.ts` (all fake data currently)
- Some changes are not related to this PR. **Do not squash and merge.**

### CODE REVIEW

File: src/app/events/Events.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export default function Events({ className }: { className?: string }) {
+    return (
+        <section className={`${className} space-y-8`}>
+            <Title />
+            {EVENTS.map((event, i) => (
+                <EventCard key={i} index={i} event={event} />
+            ))}
+        </section>
```
Comment (phoenixpereira) [inline src/app/events/Events.tsx:67]: Line up the right end of the header and the event cards.

image

File: src/app/events/Events.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    src={`/images/events/${event.image}`}
+                    alt={`${event.title}`}
+                    width={450}
+                    height={450}
+                    className="w-full shrink-0 rounded-lg border-[3px] border-black bg-white object-contain md:w-[450px]"
+                />
+                <div className="grow space-y-2 md:space-y-4">
+                    <div className="flex gap-6 font-bold">
+                        <div className="grow md:space-y-2">
+                            <h4 className="text-2xl md:border-b-[3px]  md:border-black md:pb-1 md:text-3xl">
```
Comment (phoenixpereira) [inline src/app/events/Events.tsx:34]: Remove the extra space in between `md:border-b-[3px]` and `md:border-black`.

File: src/app/events/Events.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                            </h4>
+                            <div className="flex gap-2">
+                                <FiClock size={26} />
+                                <span>{event.time}</span>
+                            </div>
+                            <div className="flex gap-2">
+                                <FiMapPin size={26} />
+                                <span>{event.location}</span>
+                            </div>
+                        </div>
```
Comment (phoenixpereira) [inline src/app/events/Events.tsx:45]: Increase the vertical gap between the title, time and location lines for mobile. Should be fixed by changing `md:space-y-2` to `space-y-2`.

File: src/app/events/page.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import Events from './Events';
+import FridayNight from './FridayNight';
+import Info from './Info';
+
+export default function EventsPage() {
+    return (
+        <main className="relative grid grid-cols-5 gap-y-12 text-lg md:text-xl">
+            <Info className="order-last col-span-5 mb-6 border-white md:order-first md:col-span-2 md:mb-0 md:border-r-4 md:pr-4" />
+            <FridayNight className="col-span-5 md:col-span-3 md:pl-8" />
+            <Events className="order-first col-span-5 md:order-last" />
```
Comment (phoenixpereira) [inline src/app/events/page.tsx:11]: On mobile, I feel like `<Info>` should be first, then `<Events>` and `<FridayNight>`.

**Review by phoenixpereira — CHANGES_REQUESTED** [review]
Just a few changes needed.

File: src/app/events/FridayNight.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+export default function FridayNight({ className }: { className?: string }) {
+    return (
+        <section className={`${className} space-y-6 md:space-y-10`}>
+            <div className="flex gap-4">
+                <Info className="grow" />
+                <Ducks />
+            </div>
+            <Details />
+        </section>
```
[the remark this answers is not in the document]
Comment (jsun969) [reply src/app/events/FridayNight.tsx:87]: This is the issue from `FancyRectangle`. ig we shouldn't fix it here.

File: src/data/events.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    | 'AUG'
+    | 'SEP'
+    | 'OCT'
+    | 'NOV'
+    | 'DEC';
+type Time = `${number}:${number}${'am' | 'pm'}`;
+export type Event = {
+    title: string;
+    date: { month: Month; day: number };
+    time: `${Time} - ${Time}`;
```
Comment (rayokamoto) [inline src/data/events.ts:20]: We should be using en dash for durations

File: src/data/events.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    | 'APR'
+    | 'MAY'
+    | 'JUN'
+    | 'JUL'
+    | 'AUG'
+    | 'SEP'
+    | 'OCT'
+    | 'NOV'
+    | 'DEC';
+type Time = `${number}:${number}${'am' | 'pm'}`;
```
Comment (rayokamoto) [inline src/data/events.ts:16]: I think we should use 24-hour time, thoughts?

**Review by rayokamoto — CHANGES_REQUESTED** [review]

File: src/data/events.ts (hunk 7 above)
Comment (jsun969) [reply src/data/events.ts:16]: `time: string` ? XD

**Review by rayokamoto — APPROVED** [review]

**Review by phoenixpereira — APPROVED** [review]

**Pull request merged**